File: Raven.Abstractions/Replication/ReplicationStatistics.cs
using System;
using System.Collections.Generic;
using Raven.Abstractions.Data;

namespace Raven.Abstractions.Replication
{
	using Raven.Json.Linq;

	public class ReplicationStatistics
	{
		public string Self { get; set; }
		public Etag MostRecentDocumentEtag { get; set; }
		public Etag MostRecentAttachmentEtag { get; set; }
		public List<DestinationStats> Stats { get; set; }

		public ReplicationStatistics()
		{
			Stats = new List<DestinationStats>();
		}
	}

	public class DestinationStats
	{
		public DestinationStats()
		{
			LastStats = new RavenJArray();
		}

		public int FailureCountInternal = 0;
		public string Url { get; set; }
		public DateTime? LastHeartbeatReceived { get; set; }
		public Etag LastEtagCheckedForReplication { get; set; }
		public Etag LastReplicatedEtag { get; set; }
		public DateTime? LastReplicatedLastModified { get; set; }
		public DateTime? LastSuccessTimestamp { get; set; }
		public DateTime? LastFailureTimestamp { get; set; }
		public int FailureCount { get { return FailureCountInternal; } }
		public string LastError { get; set; }
		public RavenJArray LastStats { get; set; }
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/replication/ReplicationStatistics.java
package net.ravendb.abstractions.replication;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import net.ravendb.abstractions.data.Etag;


public class ReplicationStatistics {

  private String self;
  private Etag mostRecentDocumentEtag;
  private Etag mostRecentAttachmentEtag;
  private List<DestinationStats> stats;

  public ReplicationStatistics() {
    this.stats = new ArrayList<>();
  }

  public String getSelf() {
    return self;
  }

  public void setSelf(String self) {
    this.self = self;
  }

  public Etag getMostRecentDocumentEtag() {
    return mostRecentDocumentEtag;
  }

  public void setMostRecentDocumentEtag(Etag mostRecentDocumentEtag) {
    this.mostRecentDocumentEtag = mostRecentDocumentEtag;
  }

  public Etag getMostRecentAttachmentEtag() {
    return mostRecentAttachmentEtag;
  }

  public void setMostRecentAttachmentEtag(Etag mostRecentAttachmentEtag) {
    this.mostRecentAttachmentEtag = mostRecentAttachmentEtag;
  }

  public List<DestinationStats> getStats() {
    return stats;
  }

  public void setStats(List<DestinationStats> stats) {
    this.stats = stats;
  }

  public class DestinationStats {

    private int failureCountInternal = 0;
    private String url;
    private Date lastHeartbeatReceived;
    private Etag lastEtagCheckedForReplication;
    private Etag lastReplicatedEtag;
    private Date lastReplicatedLastModified;
    private Date lastSuccessTimestamp;
    private Date lastFailureTimestamp;

    public int getFailureCountInternal() {
      return failureCountInternal;
    }

    public void setFailureCountInternal(int failureCountInternal) {
      this.failureCountInternal = failureCountInternal;
    }

    public String getUrl() {
      return url;
    }

    public void setUrl(String url) {
      this.url = url;
    }

    public Date getLastHeartbeatReceived() {
      return lastHeartbeatReceived;
    }

    public void setLastHeartbeatReceived(Date lastHeartbeatReceived) {
      this.lastHeartbeatReceived = lastHeartbeatReceived;
    }

    public Etag getLastEtagCheckedForReplication() {
      return lastEtagCheckedForReplication;
    }

    public void setLastEtagCheckedForReplication(Etag lastEtagCheckedForReplication) {
      this.lastEtagCheckedForReplication = lastEtagCheckedForReplication;
    }

    public Etag getLastReplicatedEtag() {
      return lastReplicatedEtag;
    }

    public void setLastReplicatedEtag(Etag lastReplicatedEtag) {
      this.lastReplicatedEtag = lastReplicatedEtag;
    }

    public Date getLastReplicatedLastModified() {
      return lastReplicatedLastModified;
    }

    public void setLastReplicatedLastModified(Date lastReplicatedLastModified) {
      this.lastReplicatedLastModified = lastReplicatedLastModified;
    }

    public Date getLastSuccessTimestamp() {
      return lastSuccessTimestamp;
    }

    public void setLastSuccessTimestamp(Date lastSuccessTimestamp) {
      this.lastSuccessTimestamp = lastSuccessTimestamp;
    }

    public Date getLastFailureTimestamp() {
      return lastFailureTimestamp;
    }

    public void setLastFailureTimestamp(Date lastFailureTimestamp) {
      this.lastFailureTimestamp = lastFailureTimestamp;
    }

    public String getLastError() {
      return lastError;
    }

    public void setLastError(String lastError) {
      this.lastError = lastError;
    }

    private String lastError;

    public int getFailureCount() {
      return this.failureCountInternal;
    }
  }

}
